router/core/operation_planner.go

```diff
@@ -53,6 +53,20 @@ func (p *OperationPlanner) preparePlan(ctx *operationContext) (*planWithMetaData
 		return nil, &reportError{report: &report}
 	}
 
```

Comment:
why do we need to do it here? could we do it during parse?

Reply:
The change I did makes it so that we match the documents name with the trimmed name for planning.

However the original query is still present, thus if reparsed would need to be manually modified, based on the context I have right now I think to make sure nothing is missed in the future when something is added, its best if this change lives in the engine (cc: @devsergiy), that way anytime a document is parsed we would set the `EndBytes` to the trimmed index for the operation name, if it exists.

Reply:
It sounds complicated. Let's choose a high enough limit (512) and return an error. Following @jensneuse approach here.